Scripts/RemediationScripts/Remediate-DontUseNonADIdentities.ps1

```diff
@@ -344,71 +364,83 @@ function Remove-AzTSNonADIdentities
         New-Item -ItemType Directory -Path $folderPath | Out-Null
     }
 
-    # Safe Check: Taking backup of Non-AD identities    
-    if ($externalAccountsRoleAssignments.length -gt 0)
+    # Safe Check: Taking backup of Non-AD identities  
+    if([String]::IsNullOrWhiteSpace($FilePath))
```

Comment:
You may want to use the `DryRun` flag for this evaluation. This could be more appropriate.

Reply:
Used $FilePath because if user is not passing file name then only we will take back up otherwise script will start taking back up for role assignments.

Reply:
`FilePath` is used only when `DryRun` is false. Hence, `DryRun` could be more appropriate.

Reply:
If DryRun = $true, then only we will take back up, if I'm understanding correct.
It's not necessary for user to use DryRun, if user wants directly to run the script to remediate, then it won't take back up in that case.
Is that expected?

Reply:
Your understanding is correct.
If `DryRun` is not used, it will either be `NULL` or `false`. 

Your comment in the script says this - `If you want to perform remediation only for DryRun output, use '-FilePath' parameter and set '-DryRun' as 'False'.` This means `DryRun` and `FilePath` are dependent on one another.

Can't we leverage this?